src/frontend/query_commands.ml

```diff
@@ -359,6 +359,49 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
     in
     List.map ~f:Mbrowse.node_loc path
 
+  | Type_definition pos ->
+    let typer = Mpipeline.typer_result pipeline in
+    let structures = Mbrowse.of_typedtree (Mtyper.get_typedtree typer) in
+    let pos = Mpipeline.get_lexing_pos pipeline pos in
+    let path = Mbrowse.enclosing pos [structures] in
+    let path =
+      let rec resolve_tlink env ty =
+        match ty.Types.desc with
+        | Tconstr (path, _, _) -> Some (env, path)
+        | Tlink ty -> resolve_tlink env ty
+        | _ -> None
```

Comment:
You should be calling `Ctypes.repr` before looking at the type desc. That will clean up `Tlink` for you.

Reply:
Thanks for the tip. I pushed a commit to address this.